src/secrules_parsing/model/secrules.tx

```diff
@@ -197,7 +197,10 @@ Macro:  INT | OperationMacro | ExtendedMacro | MacroVar;
 
 // This extended macro are a little bag of holding...
 ExtendedMacro: /[A-Za-z0-9S\-\_\.\:\ \/\%\{\}\|\+=#?]+/;
-    
+
+// macro with comma, quoted by '
+ExtendedMacroWithComma: /[A-Za-z0-9S\-\_\.\:\ \/\%\{\}\|\+=#?,]+/;
```

Comment:
Maybe a dumb question: why does the regex have to be so complex? Can't it just be (approximately) `"[^"]+"|'[^']+'`?

Reply:
It's not a dumb question, it's a valid question.

I haven't investigated it, just allowed what there is. :smile:

> Can't it just be any character within quotes and anything but comma without quotes?

Perhaps that can be. See my related comment: _we have to review the allowed arguments type at the actions..._

Note, that libmodsecurity3 uses [this](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/src/parser/seclang-scanner.ll#L434) pattern.

Reply:
They use that to find macros within the string. But I can see that we might want to parse the macro one day....

Reply:
I have a branch to decouple things a bit and make things more semantic in the future. But not for now.